diff --git a/dlls/kernel32/module.c b/dlls/kernel32/module.c
index badfe1d..a561b94 100644
--- a/dlls/kernel32/module.c
+++ b/dlls/kernel32/module.c
@@ -1108,17 +1108,21 @@ FARPROC WINAPI GetProcAddress( HMODULE hModule, LPCSTR function )
     if ((ULONG_PTR)function >> 16)
     {
         ANSI_STRING     str;
-
+TRACE("(module %p, name '%s')\n",hModule,function);
         RtlInitAnsiString( &str, function );
         nts = LdrGetProcedureAddress( hModule, &str, 0, (void**)&fp );
     }
     else
+{
+TRACE("(module %p, ordinal %d)\n",hModule,(int)function);
         nts = LdrGetProcedureAddress( hModule, NULL, LOWORD(function), (void**)&fp );
+}
     if (nts != STATUS_SUCCESS)
     {
         SetLastError( RtlNtStatusToDosError( nts ) );
         fp = NULL;
     }
+TRACE(" -> %p\n",fp);
     return fp;
 }
 
